opencensus/proto/metrics/metrics.proto

```diff
@@ -99,29 +97,15 @@ message LabelKey {
 }
 
 // A collection of data points that describes the time-varying values
-// of a gauge metric.
-message GaugeTimeSeries {
-  // The set of label values that uniquely identify this timeseries. Applies to
-  // all points. The order of label values must match that of label keys in the
-  // metric descriptor.
-  repeated LabelValue label_values = 1;
-
-  // The data points of this timeseries. Point type MUST match the MetricDescriptor.type.
-  repeated Point points = 2;
-}
-
-// A collection of data points that describes the time-varying values
-// of a cumulative metric.
-message CumulativeTimeSeries {
-  // The time that the cumulative value was reset to zero.
-  google.protobuf.Timestamp start_time = 1; // required
```

Comment:
I think we still need the start time?

Reply:
Not correct, because different label-values. Am I wrong?

Reply:
Quotes from our previous discussion:

> [X] Cumulative Point
> [X] should we hoist the start_time timestamp to time series?
> 1. Downside of doing so is that it's not relevant to GaugePoints. (Decision: have two TimeSeries types) (PR65)
> 2. Downside of not: timestamp is 16 bytes, the double value you're recording is half that.
> 3. Alternatives considered: Change Metric to have one of { repeated GaugeTimeSeries, or repeated CumulativeTimeSeries (with start_time) }

> [X] Should start_time be called reset_time? Or should we make a note that we are keeping consistent with Stackdriver terminology?
> 1. Decision: keep start_ and end_ despite perhaps not being perfect names.

So I think all Points will have the same start time, even with different label values, because start time here is actually the time when we start all the collecting and has nothing to do with single Point.

/cc @g-easy

Reply:
I don't understand, let's assume we have 2 points with the same start_timestamp (maybe even the same timestamp).

Start timestamp: 1234sec and 56ns
Current timestamp: 1234sec 59ns
{foo, bar} -> 5.2
{foo, baz} -> 4.3

In the previous model I have to create 2 different CumulativeTimeSeries:
```
CumulativeTimeSeries: {
  start_time: {1234, 56}
  label_values: [{foo, true /* has_value */}, {bar, true}]
  points: [{timestamp: {1234, 59}, value: 5.2}]
}
CumulativeTimeSeries: {
  start_time: {1234, 56}
  label_values: [{foo, true /* has_value */}, {baz, true}]
  points: [{timestamp: {1234, 59}, value: 4.3}]
}
```

In the new proposal I have to create 2 TimeSeries objects as well:
```
TimeSeries: {
  label_values: [{foo, true /* has_value */}, {bar, true}]
  points: [{start_time: {1234, 56}, timestamp: {1234, 59}, value: 5.2}]
}
TimeSeries: {
  label_values: [{foo, true /* has_value */}, {baz, true}]
  points: [{start_time: {1234, 56}, timestamp: {1234, 59}, value: 4.3}]
}
```

As you can see I cannot save any allocation or bytes on the wire, if we want to save the bytes on the wire we should move the list of labels to be part of the Point not the TimeSeries.


What I am trying to say is that 2 is not valid in the majority of the cases.

Reply:
I personally don't think that one TimeSeries is "defined" or "identify" by the start_timestamp (I know that I supported the previous decision). If we keep timestamp in the TimeSeries it means that the start_timestamp defines the TimeSeries.

Reply:
Consider cases like this:

Start timestamp: 1234sec and 56ns
1st timestamp: 1234sec 59ns
{foo, bar} -> 5.2
{foo, baz} -> 4.3

2nd timestamp: 1234sec 66ns
{foo, bar} -> 5.3
{foo, baz} -> 4.4

The previous model becomes more space efficient:
```
CumulativeTimeSeries: {
  start_time: {1234, 56}
  label_values: [{foo, true /* has_value */}, {bar, true}]
  points: [
      {timestamp: {1234, 59}, value: 5.2}, 
      {timestamp: {1234, 66}, value: 5.3}
  ]
}
CumulativeTimeSeries: {
  start_time: {1234, 56}
  label_values: [{foo, true /* has_value */}, {baz, true}]
  points: [
      {timestamp: {1234, 59}, value: 4.3}, 
      {timestamp: {1234, 66}, value: 4.4}
  ]
}
```

vs.

```
TimeSeries: {
  label_values: [{foo, true /* has_value */}, {bar, true}]
  points: [
      {start_time: {1234, 56}, timestamp: {1234, 59}, value: 5.2},
      {start_time: {1234, 56}, timestamp: {1234, 66}, value: 5.3}
  ]
}
TimeSeries: {
  label_values: [{foo, true /* has_value */}, {baz, true}]
  points: [
      {start_time: {1234, 56}, timestamp: {1234, 59}, value: 4.3},
      {start_time: {1234, 56}, timestamp: {1234, 66}, value: 4.4}
  ]
}
```

As the points for a specific set of label values accumulated, the new proposal wasted more space on the duplicated start time.

> If we keep timestamp in the TimeSeries it means that the start_timestamp defines the TimeSeries.

Yes, I think cumulative TimeSeries should be defined by the start_timestamp.

Reply:
If the common case is for a `TimeSeries` to contain one `Point` (which makes sense for a periodic exporter I guess), then it doesn't matter where `start_time` lives.

Does OpenMetrics re-transmit the same `start_time` for every cumulative `Point`?

Is this `.proto` file still a conceptual data model, or do we need to start thinking about efficiency of wire encoding?

Reply:
I need to understand what we disagree on here. If it is just where the start_time live then I can move it to TimeSeries (but this seems very strange to me) and merge the other changes first then we can argue about start_time.

Reply:
I think so - I'm fine with leaving start_time to TimeSeries and move forward with other changes.